02-intermediate-ruby/advanced-variables.md

```diff
@@ -15,137 +15,296 @@
 |:----------------:|:------------------:|------------------------------------------|
 |Local variable    | **a-z** *or* **_** | Available only within the immediate scope.
 |Instance variable | **@**              | Available to a specific instance of a class
-|Class variable    | **@@**             | Available anywhere in a specific class (including all instances) (avoid me)
-|Global variable   | **$**              | Available everywhere (don't use me).
+|Class variable    | **@@**             | Available anywhere in a specific class (including all instances) (avoid using these)
+|Global variable   | **$**              | Available everywhere (don't use these)
 |Constant          | **ALL_CAPS**       | Available in the scope of their definition
 
 ### Local Variable
-- Local variables start with a lowercase letter or an underscore.
-- Local variables are only available within the block of its initialization.
+- Local variables start with a lowercase letter or an underscore
+- Local variables are only available within the block of their initialization
 - Local variables will raise an error if they are read before they're created
 - Local variables are used often
 
+Let's examine the following code:
 ```ruby
-my_name = "<3 Jeremy!"
+class Library
+  def open_library
+    message = "The library is open"
+    puts message
+  end
+
+  def close_library
+    message = "The library is closed"
+    puts message
+  end
+end
+
+library = Library.new
+library.open_library
+library.close_library
+```
+
+Run this code by going to Terminal and running the example Ruby script: `ruby source/local_variable_example.rb` or by copying and pasting the code snippet into [RubyFiddle](http://rubyfiddle.com/).
+
+This code outputs the following:
+```
+The library is open
+The library is closed
+```
+
+Both methods use a local variable named `message`. However, the scope of `message` is limited to the methods they are defined in.
+
+Try placing the line of code `puts message` outside of the methods where it is defined (and in the `Library` definition). Running the script again should produce the following error:
+
 ```
+`<class:Library>': undefined local variable or method `message' for Library:Class (NameError)
+  from source/local_variable_example.rb:3:in `<main>'
+ ```
+
+Local variables allow you to safely assign values within a scope with full control.
 
 ### Instance Variables
 - Instance variables begin with `@`
-- Instance variables are available to the object of it's initialization
+- Instance variables are available to the object of their initialization
 - Instance variables are also used very commonly
 
 ```ruby
-class Ghost
-  attr_reader :name
+class Library
+  attr_reader :library_name
 
-  def initialize(name_var)
-    @name = name_var
+  def initialize(name)
+    @library_name = name
   end
+
 end
+
+library1 = Library.new("Woodland Library")
+library2 = Library.new("Northview Library")
+
+puts library1.library_name()
```

Comment:
I think for style it would be best to not include the `()` (since that is what we are doing in most/all other places in the curriculum)